### DISCUSSION TITLE
Pixar Toon shader in Malt

### DISCUSSION THREAD

**Discussion by ianBIT** [bnpr/Malt#296] [Q&A]
[Pixar_Toon_paper.pdf](https://github.com/bnpr/Malt/files/8672422/Pixar_Toon_paper.pdf)

https://graphics.pixar.com/library/ToonRendering/paper.pdf

hello i've been looking around trying to find a way to do this kind of shading in blender, i was told that Malt has a way to create toon lines using some kind of "differential map"  if anyone out there has any clues or idea on how to do this in blender please could you point me in the right direction, the only way i have this working at the studio at moment is using a renderer but i feel it could be done in realtime with malt maybe

**Comment by pragma37** [comment]
Sure! This is relatively straightforward to do in Malt.

![imagen](https://user-images.githubusercontent.com/36610999/167947057-72336959-f659-4fe8-b1e0-fdfa7c9b9e67.png)
![imagen](https://user-images.githubusercontent.com/36610999/167947253-886735fd-63d1-4457-838c-aa7bffee8367.png)
[line_id_mask.zip](https://github.com/bnpr/Malt/files/8673428/line_id_mask.zip)

Here's an explanation of how line rendering works in Malt:
https://github.com/bnpr/Malt/discussions/274#discussioncomment-2398023

**Reply by ianBIT** [reply]
wow! and thank you! this should keep me busy for awhile, but i have one problem, i can't seem to display the lined monkey render in my viewport like in your screenshot, i loaded your file from the zip file but nothing show up in the viewport ? any clues?

thanks again

my pc is
windows10 
rtx3090 
32mb ram
asus x570 pro motherboard

**Reply by pragma37** [reply]
Open a bug report, please:
https://github.com/bnpr/Malt/issues/new/choose

**Reply by ianBIT** [reply]
mmm, i tried the file on two other machines  -  viewport not rendering on  Intel Gigabyte Aero hdr 17 -3070 gpu moblie laptop but it is  finally working on a silmilar machine with similar specs Amd 5800x  and Msi Meg motherboard rtx 3090, so thats good news!

anyway as long as the file is working i can study the file closer
